flang-rt/lib/runtime/CMakeLists.txt

```diff
@@ -175,7 +175,7 @@ else ()
   set(f128_sources "")
 endif ()
 
-if ("${LLVM_RUNTIMES_TARGET}" MATCHES "^amdgcn|^nvptx")
+if ("${LLVM_RUNTIMES_TARGET}" MATCHES "^amdgcn|^nvptx" OR FLANG_RT_EXPERIMENTAL_OFFLOAD_SUPPORT STREQUAL "CUDA")
```

Comment:
I missed the point where the `gpu_sources` were added.  Do you know how they are being built?

Regardless of the answer, I think we need to build `supported_sources` when building for CUDA, because this is the list of files that are buildable (currently) for CUDA or OpenMP.